### ISSUE TITLE
qalculate-gtk display error on launch on High-Sierra

### ISSUE THREAD

**Issue by murraydavis** [Qalculate/qalculate-gtk#44]
Mac OSX - High Sierra, calculate-gtk is a MacPort installed in /opt/local/bin. When I try to launch qalcuate-gtk, this is what happens...

bin $ sudo ./qalculate-gtk
Password:

(qalculate-gtk:46686): Gtk-WARNING **: cannot open display:

[1]+  Stopped                 sudo env PATH=$PATH ./qalculate-gtk

Ok, so is it working? The answer is yes, since I can get help.

bin $ sudo ./qalculate-gtk --help
Usage:
  qalculate-gtk [OPTION…] EXPRESSION

Help Options:
  -h, --help                 Show help options
  --help-all                 Show all help options
  --help-gapplication        Show GApplication options
  --help-gtk                 Show GTK+ Options

Application Options:
  -n, --new-instance         Start a new instance of the application
  --display=DISPLAY          X display to use

But, how do I specify the display or even list what displays are available?

**Comment by jjstickel** [comment]
Hi,

This is Jonathan. I maintain the qalculate ports in MacPorts.

Why are you running qalculate-gtk with sudo? Try without. Also, if you 
setup MacPorts properly, it should be in your path.

Jonathan

**Comment by murraydavis** [comment]
Thank you for the response.

The sudo was my attempt to see if su could launch it. Same result as general user...

~ $ qalculate-gtk
(qalculate-gtk:703): Gtk-WARNING **: cannot open display:
[1]+  Exit 1                  qalculate-gtk

qalculate-gtk is in $PATH as you can see from above when I attempt to launch calculate. Also, note the PID 703, so the program launches, but something is wrong with the display connection. Prior to installing calculate, I updated Xcode and installed the command line tools and then updated all packages. I kept the command-line output from the qalculate install and found no errors. I also rebooted just so see if that helped, it did not.

**Comment by murraydavis** [comment]
By the way, qalc is working fine...

~ $ qalc
> 2^3
  2^3 = 8
> exit

~ $ qalc 2^3
2^3 = 8

~ $ qalc
> (x+y)^2
  (x + y)^2 = x^2 + 2xy + y^2

**Comment by jjstickel** [comment]
I'm not sure what the issue is then. Probably something related to X11 
(unless you installed gtk3 +quartz). Do you have xorg-server installed?

I suggest you post on the macports-users list, as this is more of an 
macports issue rather than a qalculate issue.

Jonathan

**Comment by murraydavis** [comment]
I was just about to update my post when I noticed your email. Yes, it is an X11 issue. I first installed the xorg-server mac-port, but that did not fix the issue. I then played around with setting the env DISPLAY variable, but that also did not work. Finally, I remembered XQuartz and installed that package. So, I launch XQuartz and then using the XQuartz Applications drop-down menu, choose terminal and then in the terminal window, type: qalculate-gtk and the calculate-gtk window appears. The font is quite poor, but I may just have to fiddle with the ZQuartz settings. Cheers! I agree, I will take a look at the macports-user lists as my issue does not appear related to calculate-gtk. Awesome software, BTW!
